### PR TITLE
fix: migrate 'api key' storage data to 'instance name' storage

### PR DESCRIPTION
**Pull request by falconandy**
### Summary

Migrate 'api key' storage data to 'instance name' storage.

Added missing logic to restore deviceId/sessionId from storage.

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-SDK-Template/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

File: Tests/AmplitudeTests/Migration/StoragePrefixMigrationTests.swift [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        XCTAssertNil(destinationLastEventId)
+
+        let destinationEventFiles = destination.getEventFiles(includeUnfinished: true)
+        XCTAssertEqual(destinationEventFiles.count, 0)
+
+        let sourceEventsStorageDirectory = source.getEventsStorageDirectory(createDirectory: false)
+        XCTAssertFalse(FileManager.default.fileExists(atPath: sourceEventsStorageDirectory.path))
+    }
+
+    func testDoNotMoveEventFilesToDestinationWithWrittenEvents() throws {
```
Comment (justin-fiedler) [inline Tests/AmplitudeTests/Migration/StoragePrefixMigrationTests.swift:97]: Do we have to drop these?

The user/session information is persisted in the event storage. So the only difference I can think of would be a change in API key, which is probably ok.

**Review by justin-fiedler — COMMENTED** [review]
Thanks @falconandy for the feature (and clean up). Overall looks good. See comments.

**Review by liuyang1520 — COMMENTED** [review]
LGTM! Thanks for updating the storage key!

File: Sources/Amplitude/Migration/StoragePrefixMigration.swift [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        moveUserDefaults()
+    }
+
+    private func moveUserDefaults() {
+        if let sourceDeviceId: String = source.read(key: StorageKey.DEVICE_ID) {
+            if destination.read(key: StorageKey.DEVICE_ID) == nil {
+                do {
+                    try destination.write(key: StorageKey.DEVICE_ID, value: sourceDeviceId)
+                } catch {
+                    print("can't write destination DEVICE_ID: \(error)")
```
[the remark this answers is not in the document]
Comment (falconandy) [reply Sources/Amplitude/Migration/StoragePrefixMigration.swift]: Done

File: Sources/Amplitude/Migration/StoragePrefixMigration.swift [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                    print("can't write destination USER_ID: \(error)")
+                }
+            }
+
+            do {
+                try source.write(key: StorageKey.USER_ID, value: nil)
+            } catch {
+                print("can't clear source USER_ID: \(error)")
+            }
+        }
```
[the remark this answers is not in the document]
Comment (falconandy) [reply Sources/Amplitude/Migration/StoragePrefixMigration.swift]: Done

File: Tests/AmplitudeTests/Migration/StoragePrefixMigrationTests.swift (hunk 1 above)
Comment (falconandy) [reply Tests/AmplitudeTests/Migration/StoragePrefixMigrationTests.swift]: The test is removed (see a comment above), new test with duplicated file names is added.

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**